x-pack/plugins/lists/public/lists/api.ts

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { chain, fromEither, map, tryCatch } from 'fp-ts/lib/TaskEither';
+import { flow } from 'fp-ts/lib/function';
+import { pipe } from 'fp-ts/lib/pipeable';
+
+import {
+  DeleteListSchemaEncoded,
+  ExportListItemQuerySchemaEncoded,
+  FindListSchemaEncoded,
+  FoundListSchema,
+  ImportListItemQuerySchemaEncoded,
+  ImportListItemSchemaEncoded,
+  ListSchema,
+  deleteListSchema,
+  exportListItemQuerySchema,
+  findListSchema,
+  foundListSchema,
+  importListItemQuerySchema,
+  importListItemSchema,
+  listSchema,
+} from '../../common/schemas';
+import { LIST_ITEM_URL, LIST_URL } from '../../common/constants';
+import { validateEither } from '../../common/siem_common_deps';
+import { toPromise } from '../common/fp_utils';
+
+import {
+  ApiParams,
+  DeleteListParams,
+  ExportListParams,
+  FindListsParams,
+  ImportListParams,
+} from './types';
+
+const findLists = async ({
+  http,
+  cursor,
+  page,
+  per_page,
+  signal,
+}: ApiParams & FindListSchemaEncoded): Promise<FoundListSchema> => {
+  return http.fetch(`${LIST_URL}/_find`, {
+    method: 'GET',
+    query: {
+      cursor,
+      page,
+      per_page,
+    },
+    signal,
+  });
+};
+
+const findListsWithValidation = async ({
+  http,
+  pageIndex,
+  pageSize,
+  signal,
+}: FindListsParams): Promise<FoundListSchema> =>
+  pipe(
```

Comment:
Love this. I did validations on the exceptions, but it's way more verbose. I'd opt to update the exceptions list to use these same patterns.